generator/gen_bindings.go

```diff
@@ -251,10 +248,7 @@ func (gen *Generator) unpackSlice(buf1 io.Writer, buf2 *reverseBuffer, cgoSpec t
 		gen.submitHelper(sizeOfPtr)
 		gen.submitHelper(cgoAllocMap)
 
-		fmt.Fprintf(buf1, `allocs = new(cgoAllocMap)
-		defer runtime.SetFinalizer(&unpacked, func(*%s) {
```

Comment:
How this removal is compensated? There are some cases like media player where packets of `[][][]float32` are being allocated per second.

Reply:
this change is for the 1st issue, that gc caused fatal address inside the `packSGogcPublicReplicaInfo` call

since functions like unpackXXX & packXXX are private, i think that we could just focus on the uintptr's lifetime, thus the unpack/pack cycle.

we should keep the uintptr alive until the pack things are completed.
but adding a single `runtime.KeepAlive` of the uintptr, and leaving the `SetFinalizer` untouched seems to be useless in my tests.

so, i removed these SetFinalizers here, and added a `defer freeing` of the returned allocMap after every unpack call, right  in the arg proxy  logic.

pls correct me if i misunderstood anything~~

Reply:
i also made some nested slices to be generated by my patch,  to see if it would cause some troubles

[see this](https://github.com/dtynn/go-gc/blob/test/c-for-go/gen/cgo_helpers.go#L182-L219)

i think that all the allocations are tracked by the root alloc map, am i right on this?